halo/cmd/ready.go

```diff
@@ -0,0 +1,61 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/log"
+
+	cmtcfg "github.com/cometbft/cometbft/config"
+
+	"github.com/spf13/cobra"
+)
+
+func newReadyCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "ready",
+		Short: "Assert the readiness of the halo node",
+		Args:  cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			err := assertReady(cmd.Context())
+			if err != nil {
+				return errors.Wrap(err, "ready failed")
+			}
+
+			return nil
+		},
+	}
+
+	return cmd
+}
+
+// assertReady calls halo's /ready endpoint and returns nil if the status is ready
+// or an error otherwise.
+func assertReady(ctx context.Context) error {
+	cfg := cmtcfg.DefaultConfig()
+	url := fmt.Sprintf("http://0.0.0.0%v/ready", cfg.Instrumentation.PrometheusListenAddr)
+
```

Comment:
suggest rather:
```
host, port, err := net.SplitHostPort(readyCfg.MonitoringAddr)
if err ...
else if host == "" {
  host = "locahost"
}

addr := net.JoinHostPort(host, port)
```

Reply:
can also just simply this by setting default `monitoringAddr="http://localhost:26660"`

Reply:
Yes, but I'd still take the port from `cfg.Instrumentation.PrometheusListenAddr`

Reply:
It is a tradeoff of readability vs correctness. I'd stick with simple string literal. We can always add a test that ensures the string literal matches `cfg.Instrumentation.PrometheusListenAddr`

Reply:
Sounds good!